### ISSUE TITLE
Capture does not detect eye-tracker microphone - UBUNTU

### ISSUE THREAD

**Issue by fwalker89** [pupil-labs/pupil#247]
Dear all,

The world camera of the Pupil Pro has a built in microphone. When I select this microphone as audio source on Pupil Capture, it still keeps using the microphone of the laptop. This really doesn't work out for me since, when I record, the laptop is closed and placed in a backpack. I also tried to bypass the problem by using a non-usb external microphone. This also doesn't work.

This problem doesn't occur when I play Capture on Mac, but it does when I run it on Ubuntu.

Hope you guys have a solution.

Thank you,
Francesco

**Comment by dschreij** [comment]
The Ubuntu version was 15.04 by the way. Thanks!

**Comment by cpicanco** [comment]
Hi, I don't know if it is an OS specific issue. I would like to note that the "official support" of pupil is for Ubuntu 14.04 LTS. Hopefully, it will not be OS specific.

**Comment by mkassner** [comment]
I ll look into that!

**Comment by mkassner** [comment]
I just did a test on 14.04 and could not reproduce this error. I can record form both the headset microphone and internal microphone. 

What happens when you do this:

```
arecord -l
```

and then 

```
ffmpeg -f alsa -i plughw:1 -t 10 out.wav
```

where you replace the '1' with the id for the correct microphone.
This should record 10s from the correct input, we use something similar internally.

Best,

M

**Comment by dschreij** [comment]
Francesco, could you please have a look at this?

**Comment by mkassner** [comment]
Hi! 

Any news on this?

**Comment by fwalker89** [comment]
Version of the recorder is 0.6.170 btw.


We can do this correctly from command line, but not from  Pupil Capture. Capture keeps detecting the internal mic even if audio_src c930e (Logitech Webcam c93) is selected.

**Comment by mkassner** [comment]
Hi,

could you post the output from `arecord -l` ?

M

**Comment by fwalker89** [comment]
This gives the following output:

```
**** List of CAPTURE Hardware Devices ****
card 1: PCH [HDA Intel PCH], device 0: ALC283 Analog [ALC283 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: C930e [Logitech Webcam C930e], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

**Comment by mkassner** [comment]
Aha!

I get this:
```python
**** List of CAPTURE Hardware Devices ****
card 0: AudioPCI [Ensoniq AudioPCI], device 0: ES1371/1 [ES1371 DAC2/ADC]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: C930e [Logitech Webcam C930e], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
```

The indices you have are not 0 based for some reason. I just pushed a fix.
Can you pull and test?

M

**Comment by mkassner** [comment]
Any updates? Where you able to test this?

**Comment by dschreij** [comment]
No sorry not yet (We were actually doing a project with the pupil eyetracker in the Van Gogh museum the past few days), but I am going to test this now. Should I test from source or is there a new release (.deb file) which I can use. I still have trouble getting the pupil software to run from source, but I can attempt it again.

**Comment by mkassner** [comment]
Hi Daniel,

I can bundle and upload.

give me 24hrs.

M

**Comment by mkassner** [comment]
https://github.com/pupil-labs/pupil/releases/tag/v0.6.19

**Comment by dschreij** [comment]
Thanks, I will get Francesco to check this and get back to you a.s.a.p!

**Comment by dschreij** [comment]
On my machine (Ubuntu 15.10 with an AMD HD6450) I am getting the following traceback when trying to load a pupil recording folder with player 0.6.19

```
libGL error: unable to load driver: r600_dri.so
libGL error: driver pointer missing
libGL error: failed to load driver: r600
libGL error: unable to load driver: swrast_dri.so
libGL error: failed to load driver: swrast
Traceback (most recent call last):
  File "<string>", line 535, in <module>
  File "<string>", line 224, in session
  File "pyglui/cygl/utils.pyx", line 45, in pyglui.cygl.utils.init (pyglui/cygl/utils.c:3021)
  File "pyglui/cygl/utils.pyx", line 56, in pyglui.cygl.utils.init (pyglui/cygl/utils.c:2967)
Exception: This OpenGL context is below 2.1.
```
The strange thing is that pupil player normally starts when not passing a folder as an argument...

**Comment by mkassner** [comment]
Hi,

so it works but only when stating without args and dragging the recording
onto the window?

This error usually means that pupil is using the wrong libc++ lib but we
have stopped shipping it with the bundle two versions ago.

M

**Comment by dschreij** [comment]
It also affects pupil-capture. This doesn't start at all.

This is what I could find about my libc++ files:

```
$ dpkg -l | grep c++
rc  libflac++6:amd64                                      1.3.1-1ubuntu1                             amd64        Free Lossless Audio Codec - C++ runtime library
ii  libflac++6v5:amd64                                    1.3.1-4                                    amd64        Free Lossless Audio Codec - C++ runtime library
rc  libsigc++-2.0-0c2a:amd64                              2.4.0-1                                    amd64        type-safe Signal Framework for C++ - runtime
ii  libsigc++-2.0-0v5:amd64                               2.4.1-2                                    amd64        type-safe Signal Framework for C++ - runtime
ii  libstdc++-4.8-dev:amd64                               4.8.5-1ubuntu1                             amd64        GNU Standard C++ Library v3 (development files)
ii  libstdc++-4.9-dev:amd64                               4.9.3-5ubuntu1                             amd64        GNU Standard C++ Library v3 (development files)
ii  libstdc++-5-dev:amd64                                 5.2.1-22ubuntu2                            amd64        GNU Standard C++ Library v3 (development files)
ii  libstdc++6:amd64                                      5.2.1-22ubuntu2                            amd64        GNU Standard C++ Library v3
ii  libstdc++6:i386          
```

**Comment by mkassner** [comment]
Hi,

I m not sure if this is related but I only have `libstdc++6` not any older
versions.

Could you also try running capture and player from terminal using
`pupil-capture` and `pupil-player`?

**Comment by mkassner** [comment]
Any news on wether the audio works for you? I tried running the latest bundle on a Xubuntu 15.04 live stick and audio worked fine for both internal and headset microphone. (This was using a ms surface pro 4).

**Comment by dschreij** [comment]
I removed the older libc++ libraries on my computer and still get the same error. The error messages I posted above are when pupil_capture and _player are being run from the terminal. Otherwise the app does not start up and gives no error message at all.
 We are a bit apprehensive at the moment to upgrade our pupil version on the laptop which we have reserved for pupil eye tracking  (and which shows the audio error), because we are going to perform an experiment with it again this Friday evening. I'm willing to test the newer version after this day if this is ok, as we have no further research planned with pupil and I will cause no problems if I accidentally mess up the laptop with updates.
